### ISSUE TITLE
Documentation Overhaul

### ISSUE THREAD

**Issue by shortercode** [shortercode/Doom.js#18]
So I've been thinking Doom is kinda lacking in the documentation front, sometimes I forget what it's full feature set is! The landing page predates the web timing project (i think), and as such is only a very basic description. It is surprisingly still fairly accurate though.

We could do with a fairly simple page for each method describing the parameters and effects, should make the project a lot more accessible!

**Comment by jcf120** [comment]
Yeah, would be good. Recently Allan (who joined shortly after you left) and Michael (since summer) mentioned they'd like more Docs. How d'you wanna go about this? Gen tool?

**Comment by shortercode** [comment]
Mmm maybe? I haven't really used any before, any you know that are good? I'm not fond of codebases that generate docs from special comments, I find it harder to overview files with many large comments. Also we have a slightly non conventional structure

**Comment by AJ-Moore** [comment]
I think it would be nice to follow a style similar to the original landing page doc, I feel it doesn't necessarily have to be very complicated just provide some additional bare minimalistic examples to cover some of the newer features alloys/ reference mapping etc... additional to as you said before providing a fairly simplistic page/ description  for each method/ parameter and the effect it has.

**Comment by shortercode** [comment]
Would certainly make life a little easier to write something like that, bit less formal. Still be good to get some of the more obscure behaviours in though. So far I think the basic categories I've come up with are:

- Doom.create
- Doom.modify
- Doom.fetch
- Doom.search
- Doom.remove
- partials
- mutation events
- alloys
- reference maps
- touch gestures

Think it makes sense to write each on up separate, then can edit the smaller/similar ones together. Going to make a branch and start writing a couple of them. Be nice to get them done quickly, hoping to get a load of my personal projects done over the xmas break!

**Comment by AJ-Moore** [comment]
Cool
